content/en/real_user_monitoring/error_tracking/mobile/ios.md

```diff
@@ -100,6 +100,51 @@ Datadog.initialize(
 CrashReporting.enable()
 ```
 
+### Add app hang reporting
+
+App hangs are an iOS-specific type of error that happens whenever the application is being unresponsive for too long.
+
+By default, app hangs reporting is **disabled**, but you can enable it and set your own threshold to monitor app hangs that last more than a specified duration using the `appHangThreshold` initialization parameter. A common threshold for reporting an app hang is between 2 and 3 seconds. However, having a customizable threshold makes it easier to find the right balance between fine-grained and noisy observability.
+
+App hangs are reported through the RUM SDK (not through Logs).
+
+When **enabled**, any main thread pause that is longer than the specified `appHangThreshold` is considered a "hang" in the Error Tracking page.
+
+- **Fatal app hang** - How a hang gets reported if it never gets recovered and the app is terminated. Fatal app hangs are marked as "Crash" on them in the Error Tracking page.
+
+  {{< img src="real_user_monitoring/error_tracking/ios-fatal-app-hang.png" alt="A fatal app hang in the Error Tracking page." style="width:60%;" >}}
+
+- **Non-fatal app hang** - How a hang gets reported if the app recovers from a relatively short hang and continues running. Non-fatal app hangs do not have a "Crash" mark on them in the Error Tracking page.
+
+  {{< img src="real_user_monitoring/error_tracking/ios-non-fatal-app-hang.png" alt="A non-fatal app hang in the Error Tracking page." style="width:60%;" >}}
+
+#### Enable app hangs monitoring
+
+To enable app hangs monitoring:
+
+1. During or after the [instrumentation][13] step of the RUM SDK, update the initialization snippet with the `appHangThreshold` parameter:
+
+   ```swift
+   RUM.enable(
+       with: RUM.Configuration(
+           applicationID: "<rum application id>",
+           appHangThreshold: 0.25
+       )
+   )
+   ```
+
+2. Set the `appHangThreshold` parameter to the minimal duration you want app hangs to be reported. For example, enter `0.25` to report hangs lasting at least 250 ms.
+
+**Note**: The minimum value this option can be set to is `0.1` seconds (100 ms). However, setting the threshold to such small values may lead to an excessive reporting of hangs. The SDK implements a secondary thread for monitoring app hangs. To reduce CPU utilization, it tracks hangs with a tolerance of 2.5%, which means some hangs that last close to this threshold may not be reported.
+
+**Note**: Apple only considers hangs lasting more than 250 ms in their hang rate metrics in Xcode Organizer. Datadog recommends starting with a similar value for the `appHangThreshold` (in other words, set it to `0.25`) and lowering it incrementally if you feel you are missing out on observability.
```

Comment:
Here we recommend to start with 0.25 and lower it if they feel missing out on observability. My gut feeling is that `0.25s` will be in fact super noisy, so people will rather want to higher it rather than lower.

@MaelNamNam what's your take on this?

Reply:
I'd rather stick to `0.25` as [this is the threshold used by Apple in Xcode Organizer](https://developer.apple.com/documentation/xcode/analyzing-responsiveness-issues-in-your-shipping-app#View-your-apps-hang-rate:~:text=of%20unresponsiveness%20of-,more%20than%20250%20ms.,-The%20Organizer%20window)